cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -269,6 +276,41 @@ class device_row_comparator {
   std::optional<device_span<null_order const>> const _null_precedence;
 };  // class device_row_comparator
 
+/**
+ * @brief Wraps and interprets the result of templated Comparator that returns a weak_ordering.
+ * Returns true if the weak_ordering matches any of the templated values.
+ *
+ * Note that this should never be used with only `weak_ordering::EQUIVALENT`.
+ * An equality comparator should be used instead for optimal performance.
+ *
+ * @tparam Comparator generic comparator that returns a weak_ordering.
+ * @tparam values weak_ordering parameter pack of orderings to interpret as true
+ */
+template <typename Comparator, weak_ordering... values>
+struct weak_ordering_comparator_impl {
+  __device__ bool operator()(size_type const& lhs, size_type const& rhs)
+  {
+    weak_ordering const result = comparator(lhs, rhs);
+    return ((result == values) || ...);
+  }
+  Comparator comparator;
+};
+
+/**
+ * @brief Wraps and interprets the result of device_row_comparator, true if the result is
+ * weak_ordering::LESS meaning one row is lexicographically *less* than another row.
+ *
+ * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
+ */
+template <typename Nullate>
+using less_comparator =
+  weak_ordering_comparator_impl<device_row_comparator<Nullate>, weak_ordering::LESS>;
+
+template <typename Nullate>
+using less_equivalent_comparator = weak_ordering_comparator_impl<device_row_comparator<Nullate>,
+                                                                 weak_ordering::LESS,
+                                                                 weak_ordering::EQUIVALENT>;
+
```

Comment:
It would be useful to have aliases for the remaining comparator too (i.e., `>` and `>=`).

Reply:
https://github.com/rapidsai/cudf/pull/9452 uses `less_equivalent_comparator`.